### PR TITLE
jenkins job and script for cloudflare

### PR DESCRIPTION
**Pull request by syed-awais-ali**
@edx/devops Review please. Thanks

### CODE REVIEW

File: devops/jobs/CloudFlareHitRate.groovy [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+*/
+package devops.jobs
+
+
+import static org.edx.jenkins.dsl.DevopsTasks.common_parameters
+
+
+class CloudflareHitRate {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('ZONE_ID') : "Zone id of domain"
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: The assertion errors should be more clear about the error. eg. `Required ZONE_ID setting missing from configuration.`

Similarly for the two assertions below as well.

File: devops/jobs/CloudFlareHitRate.groovy [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/cloudflare-hit-rate-edx") {
+
+
+            def zone_id = extraVars.get('ZONE_ID')
+            def auth_key = extraVars.get('AUTH_KEY')
+            def email = extraVars.get('EMAIL')
+
+
+            triggers {
+                cron("H/8 * * * *")
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: Why every 8 minutes?  Did you mean for this to run every 8 hours?

File: devops/jobs/CloudFlareHitRate.groovy [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+package devops.jobs
+
+
+import static org.edx.jenkins.dsl.DevopsTasks.common_parameters
+
+
+class CloudflareHitRate {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('ZONE_ID') : "Required ZONE_ID setting missing from configuration"
+        assert extraVars.containsKey('AUTH_KEY') : "Required Authentication key setting missing from configuration"
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: You should put the name of the key in the error message.  `Required authentication key(AUTH_KEY) setting missing from configuration.`

Similarly for the EMAIL key below.

File: devops/jobs/CloudFlareHitRate.groovy [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+class CloudflareHitRate {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('ZONE_ID') : "Required ZONE_ID setting missing from configuration"
+        assert extraVars.containsKey('AUTH_KEY') : "Required Authentication key(AUTH_KEY) setting missing from configuration"
+        assert extraVars.containsKey('EMAIL') : "Required email(EMAIL) setting missing from configuration"
+        dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/cloudflare-hit-rate-edx") {
+
+
+            def zone_id = extraVars.get('ZONE_ID')
+            def auth_key = extraVars.get('AUTH_KEY')
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: This shouldn't come from extraVars it should come from a credential.

[Look here](https://github.com/edx/jenkins-job-dsl/blob/master/devops/jobs/ManageNewRelicServerAlerts.groovy#L32-L36) for an example.

File: devops/resources/cloudflare-hit-rate.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -exuo pipefail
+
+cd $WORKSPACE/configuration
+env
+
+python util/jenkins/cloudflare-hit-rate-monitor.py\
```
Comment (feanil) [inline devops/resources/cloudflare-hit-rate.sh]: Don't we need to pass in a hit threshold?

File: devops/jobs/CloudFlareHitRate.groovy [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import static org.edx.jenkins.dsl.DevopsTasks.common_parameters
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+
+
+class CloudflareHitRate {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('ZONE_ID') : "Required ZONE_ID setting missing from configuration"
+        assert extraVars.containsKey('EMAIL') : "Required email(EMAIL) setting missing from configuration"
+        dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/cloudflare-hit-rate-edx") {
+            wrappers common_wrappers
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: Please also include `logRotator common_logrotator`  we don't want to keep the logs forever.

File: devops/jobs/CloudFlareHitRate.groovy [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+                    extensions {
+                        cleanAfterCheckout()
+                        pruneBranches()
+                        relativeTargetDirectory('configuration')
+                    }
+                }
+            }
+
+            parameters common_parameters(extraVars)
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: You don't need all these parameters if the only repo you're cloning is configuration.  Just include the parameters for the configuration repo.

**Comment by syed-awais-ali** [comment]
@feanil Done.

File: devops/jobs/CloudFlareHitRate.groovy [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        * SECURE_GIT_CREDENTIALS: (Required - jenkins name of git credentials)
+        * ZONE_ID: (Required - The CloudFlare Zone ID)
+        * AUTH_KEY: (Required - Authentication key for the account that would make API calls)
+        * EMAIL: (Required - Email of the account for making API calls)
+
+*/
+package devops.jobs
+
+
+import static org.edx.jenkins.dsl.DevopsTasks.common_parameters
```
Comment (feanil) [inline devops/jobs/CloudFlareHitRate.groovy]: This is no longer used.

**Pull request merged**